web_ui/frontend/components/layout/Sidebar.tsx

```diff
@@ -20,12 +20,141 @@
 
 import Image from 'next/image'
 import { useRouter } from 'next/navigation'
-import {Typography, Box, Button, Snackbar, Alert, Tooltip} from "@mui/material";
+import {Typography, Box, Button, Snackbar, Alert, Tooltip, IconButton, Grow, Paper, BoxProps} from "@mui/material";
+import { ClickAwayListener } from '@mui/base';
+import {Help, HelpOutline, Description, BugReport} from "@mui/icons-material";
 import LogoutIcon from '@mui/icons-material/Logout';
 
 import styles from "../../app/page.module.css"
-import GithubIcon from "../../public/static/images/github-mark.png"
+import GitHubIcon from '@mui/icons-material/GitHub';
 import React, {ReactNode, useState} from "react";
+import Link from 'next/link';
+
+interface SpeedButtonProps {
+    open: boolean,
+    order: number,
+    icon: ReactNode,
+    title: string,
+    onClick?: () => void
+    href?: string
+    boxProps?: BoxProps
+}
+
+export const getVersionNumber = () => {
+    const { version } = require('../../package.json');
+    return version;
+}
+
+const SpeedDialButton = ({open, order, icon, title, onClick, href, boxProps} : SpeedButtonProps) => {
+
+    // Logical XOR
+    if((href != undefined) == (onClick != undefined)){
+        throw new Error("SpeedDialButton must have either an onClick xor href prop")
+    }
+
+    return (
+        <Grow
+            in={open}
+            style={{ transformOrigin: '0 0 0' }}
+            {...(open ? { timeout: 200 * order } : {})}
+        >
+            <Box pl={1} {...boxProps}>
+                <Tooltip title={title} arrow>
+                    <Paper elevation={2} sx={{ borderRadius: "50%", bgcolor: "#ffffff00"}}>
+                        { href != undefined ?
+                            <Link href={href} target={"_blank"}>
```

Comment:
Do you think it's still worth adding `rel="noopener noreferrer"` to this? I just know that most of the latest browsers add `rel="noopener"` when `target="_blank"`